prometheus_client/exposition.py

```diff
@@ -70,10 +71,18 @@ def generate_latest(registry=core.REGISTRY):
 
 class MetricsHandler(BaseHTTPRequestHandler):
     def do_GET(self):
+        try:
+            content = generate_latest(core.REGISTRY)
+        except:
+            logging.exception('error generating response')
```

Comment:
Please don't use the "convenience" functions directly on the logging module. They log to the root logger making it impossible to filter (or even redirect) these messages. Instead use a module level logger like so:

```
import logging

log = logging.getLogger(__name__)

# ... code ...

log.exception("whatever")

```